api/services/okta_service.py

```diff
@@ -85,147 +87,187 @@ async def _retry(func: Callable[[Any], Any], *args: Any, **kwargs: Any) -> Any:
             await asyncio.sleep(wait_time)
 
     def get_user(self, userId: str) -> User:
-        user, _, error = asyncio.run(OktaService._retry(self.okta_client.get_user, userId))
+        async def _get_user(userId: str) -> User:
+            request_executor: OktaRequestExecutor = self.okta_client.get_request_executor()
```

Comment:
Can this block be abstracted into a separate method or moved into the `OktaService._retry()` helper to cut down a bit on the redundancy?


```
request_executor: OktaRequestExecutor = self.okta_client.get_request_executor()
async with aiohttp.ClientSession() as session:
    request_executor.set_session(session)
```

Reply:
I separated it from the `_retry` helper so that paginated operations could share a connection, but took your advice to create a context manager and helper that handles sessions.